src/operator.rs

```diff
@@ -903,6 +905,62 @@ impl UnaryPreAggOperator for Total {
     }
 }
 
+pub struct SplitDef {
+    separator: String,
+    input_column: Option<String>,
+    output_column: String,
+}
+
+impl SplitDef {
+    pub fn new(separator: String, input_column: Option<String>, output_column: String) -> Self {
+        SplitDef {
+            separator,
+            input_column,
+            output_column,
+        }
+    }
+}
+
+impl OperatorBuilder for SplitDef {
+    fn build(&self) -> Box<dyn UnaryPreAggOperator> {
+        Box::new(Split::new(
+            self.separator.clone(),
+            self.input_column.clone(),
+            self.output_column.clone(),
+        ))
+    }
+}
+
+pub struct Split {
+    separator: String,
+    input_column: Option<Expr>,
+    output_column: String,
+}
+
+impl Split {
+    pub fn new(separator: String, input_column: Option<String>, output_column: String) -> Self {
+        Self {
+            separator,
+            input_column: input_column.map(Expr::Column),
+            output_column,
+        }
+    }
+}
+
+impl UnaryPreAggOperator for Split {
```

Comment:
since split is stateless it's actually `UnaryPreAggFunction` (these names are getting out of hand and need documentation) -- this will mean that you don't need to go through the `LimitDef` rigmarole. `split` is less like `limit` and more like `ParseLogfmt`

Reply:
You'll note the difference is whether the process method you write has `&mut self` (which we don't need) and `&self` (for stateless operators).

Reply:
Ah okay, `&self` makes sense. Changed to `UnaryPreAggFunction`